fl4health/strategies/feddg_ga_strategy.py

```diff
@@ -322,45 +335,54 @@ def update_weights_by_ga(self, server_round: int, cids: List[str]) -> None:
             server_round: (int) the current server round.
             cids: (List[str]) the list of client ids that participated in this round.
         """
-        value_list = []
-        # calculating local vs global metric difference
+        generalization_gaps = []
+        # calculating local vs global metric difference (generalization gaps)
         for cid in cids:
             assert (
                 cid in self.train_metrics and cid in self.evaluation_metrics
             ), f"{cid} not in {self.train_metrics.keys()} or {self.evaluation_metrics.keys()}"
 
+            assert self.fairness_metric.metric_name is not None
+
             global_model_metric_value = self.evaluation_metrics[cid][self.fairness_metric.metric_name]
             local_model_metric_value = self.train_metrics[cid][self.fairness_metric.metric_name]
             assert isinstance(global_model_metric_value, float) and isinstance(local_model_metric_value, float)
 
-            value_list.append(global_model_metric_value - local_model_metric_value)
+            generalization_gaps.append(global_model_metric_value - local_model_metric_value)
 
-        # calculating norm gap
-        value_list_ndarray = np.array(value_list)
-        max_value = np.max(np.abs(value_list_ndarray))
+        # Calculating the normalized the generalization gaps
+        generalization_gaps_ndarray = np.array(generalization_gaps)
+        mean_generalization_gap = np.mean(generalization_gaps_ndarray)
+        var_generalization_gaps = generalization_gaps_ndarray - mean_generalization_gap
+        max_var_generalization_gap = np.max(np.abs(var_generalization_gaps))
 
-        if max_value == 0:
+        if max_var_generalization_gap == 0:
             log(
                 WARNING,
-                "Max value in metric diff list is 0. Adjustment weights will remain the same. "
-                + f"Value list: {value_list}",
+                "Max variance in generalization gap is 0. Adjustment weights will remain the same. "
+                + f"Generalization gaps: {generalization_gaps}",
             )
-            norm_gap_list = np.zeros(len(value_list))
+            normalized_generalization_gaps = np.zeros(len(generalization_gaps))
```

Comment:
Super minor, but rather than doing `np.zeros(len(generalization_gaps)` you can simplify it as `np.zeros_like(generalization_gaps)`